src/object/obj_rpc.c

```diff
@@ -344,6 +349,14 @@ crt_proc_daos_iod_t(crt_proc_t proc, crt_proc_op_t proc_op, daos_iod_t *dvi,
 				D_GOTO(free, rc = -DER_NOMEM);
 		}
 	}
+	if (existing_flags & IOD_KCSUM_EXIST) {
+		rc = crt_proc_daos_csum_buf_t(proc, dvi->iod_kcsum);
```

Comment:
this is good, to reduce the RPC size as far as possible. we may need similar method for the added orw_dkey_csum as comment below.